backend/tests_utils.py

```diff
@@ -32,92 +41,75 @@ def assertStatusCodeIn(self, res, status_codes, user_msg=""):
 
-        password = "password"
-
-        if admin:
-            return User.objects.create_superuser(
-                username,
-                f"Admin_{username}",
-                "User",
-                f"admin_{username}@mines-paristech.fr",
-                password,
-                "2018-06-06",
-                2018,
-            )
-        else:
-            return User.objects.create_user(
-                username,
-                f"Simple_{username}",
-                "User",
-                f"simple_{username}@mines-paristech.fr",
-                password,
-                "2018-06-06",
-                2018,
-            )
-
-    def create_and_login_user(self, username="user", admin=False):
-        """
-        Create a dummy user with the specified rights, log it in and create the JWT token.
+    def options(self, url, data=None):
+        return self.client.options(self.api_base + url, data)
 
-        :param str username: the user's id.
-        :param bool admin: if True, the user will be an admin; otherwise, it will be a simple user.
-        :return: the created user
-        """
 
-        user = self.create_user(username, admin)
-        self.login(username, "password")
-        return user
+class WeakAuthenticationBaseTestCase(BaseTestCase):
+    """
+        This test base provides convenient methods to log users in and out.
+        However, it does so by generating fake JWT and disabling the signature verification.
+        Do not use this test base to test the JWT authentication itself.
+    """
 
-    def get(self, url, data=None):
-        return self.client.get(self.api_base + url, data)
+    def setUp(self):
+        settings.JWT_AUTH_SETTINGS["VERIFY_SIGNATURE"] = False
+
+    def _get_fake_jwt(self, username):
+        """
+            Produce a fake JWT authenticating the given user.
+            Use a random RSA private key.
+        """
 
-    def post(self, url, data=None, format="json", content_type="application/json"):
-        return self.client.post(
-            self.api_base + url, data, format=format, content_type=content_type
-        )
+        payload = {
+            "exp": int(datetime(2100, 1, 1).timestamp()),
+            "iss": settings.JWT_AUTH_SETTINGS["ISSUER"],
+            "aud": "portail",
+            "jti": "ec1901b49bae43f8b3e90782b0c8b131",
+            settings.JWT_AUTH_SETTINGS["USER_ID_CLAIM_NAME"]: username,
+        }
+
+        return jwt.encode(
+            payload=payload,
+            key=FAKE_PRIVATE_KEY,
+            algorithm=settings.JWT_AUTH_SETTINGS["ALGORITHM"],
+        ).decode(
+            "utf-8"
+        )  # jwt.encode returns a bytes object, it thus has to be decoded to a str.
 
-    def patch(self, url, data=None, format="json", content_type="application/json"):
-        return self.client.patch(
-            self.api_base + url, data, format=format, content_type=content_type
-        )
+    def login(self, username, password="password"):
```

Comment:
We may not need the `password` keyword now

Reply:
OK, I will just check if it doesn't break any test

Reply:
It was indeed useless